### ISSUE TITLE
(bug): iOS - Phone "type" is always "_$!<Other>!$_"

### ISSUE THREAD

**Issue by homerlex** [EinfachHans/cordova-plugin-contacts-x#9]
# Bug Report

**Current Behavior:**
When retrieving contacts on iOS I see the "type" property for every phone number is `_$!<Other>!$_`.  In Android the type is properly set to values like "home", "mobile", etc.

**Expected Behavior:**
When retrieving contacts on iOS I see the "type" property for every phone number should be set to values like "home", "mobile", etc. (as they appear when viewing contacts in iOS)

**Steps to reproduce:**
Just do a `window.ContactsX.find()` in iOS and inspect the results.

**Screenshots:**
N/A

**Environment information:**

- Cordova CLI version 
	- 9.0.0
- Cordova platform version
	- ios 6.1.1
- Plugins & versions installed in project (including this plugin)
    - cordova-plugin-add-swift-support 2.0.2 "AddSwiftSupport"
    - cordova-plugin-contacts-x 2.0.0 "ContactsX"
    - cordova-plugin-whitelist 1.3.4 "Whitelist"
- Dev machine OS and version, e.g.
    - OSX
        - MacOSX v10.15.7 build 19H524

**Comment by EinfachHans** [comment]
It works for me, wit my contacts, can you in the first step please create a reproducible repo?

**Comment by homerlex** [comment]
I can (will later) but there is nothing special.  I created a new cordova app, added this plugin, and put the following code in the onDeviceReady function:

```
        this.receivedEvent('deviceready');

        window.ContactsX.requestPermission(function(success) {
            console.log("permission success!" ,success);
        }, function (error) {
            console.error("permission error" ,error);
        });

        window.ContactsX.find(function(success) {
            console.log("find success!" , success);
        }, function (error) {
            console.error("find error" ,error);
        }, {
            fields: {
                phoneNumbers: true,
                emails: true
            }
        });
    },
```

FYI - the types work properly for emails, just not phone numbers.  I have attached a screenshot of the console output.
![contacts-x](https://user-images.githubusercontent.com/296630/109962999-e6ecba80-7cb9-11eb-8862-f2cc0c7e46b1.png)

I tested on an iPad Pro (model MLMP2LL/A) with iOS version 14.4.  I also tested on an old iPhone 5 with iOS version 12.5

**Comment by EinfachHans** [comment]
If there are real phone numbers in that screenshot, pls remove the image as this is sensitive data and this thread is public^^ i want to check if this is related to code or maybe contacts. Please follow up with the repo 😊

**Comment by homerlex** [comment]
@EinfachHans - FYI - those are fake numbers

**Comment by homerlex** [comment]
@EinfachHans - The following is a repo I created.  For this one I only installed the iOS platform.  It was created on a different machine using Cordovo 10 and iOS platform version 6.1.0.  The result is the same with this version as my previous version when getting contacts off my previously mentioned devices.

https://github.com/homerlex/hello-cordova-contacts

When you run the app you will need to look at the console log for the results.

If you cannot repro with this repo and you need me to puts some debugs in somewhere let me know.  I'd be happy to help debug.  I just need to know where you need me to look.

**Comment by EinfachHans** [comment]
To keep you updated: I'm currently so busy that i don't find time yet. If you are into native development a PR would be great!

**Comment by homerlex** [comment]
@EinfachHans - no worries, our use case doesn't require the types anyway.  I'm not an iOS native developer.  If the problem were in Android I'd take a look :-)

**Comment by Mickey34** [comment]
@homerlex 
The problem in the getPhoneNumber():
```
    func getPhoneNumbers() -> [NSDictionary] {
        let labeledValues: [NSDictionary] = self.contact.phoneNumbers.map { (ob: CNLabeledValue<CNPhoneNumber>) -> NSDictionary in
            return [
                "id": ob.identifier,
                "type": ContactsX.mapStringToLabel(string: ob.label ?? ""),
                "value": ob.value.stringValue
            ]
        }
        return labeledValues;
    }
```
There should be ContactsX.mapLabelToSring(label: ob.label ?? "") instead of ContactsX.mapStringToLabel(string: ob.label ?? "")
Careful, there is also a typo in the function name an you should check that you typed correct name (Sring instead of String).

**Comment by EinfachHans** [comment]
PR's are welcome

**Comment by EinfachHans** [comment]
Should be fixed, pls retest in `2.0.1`
